test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
> the line `palindromModel.reviewObj = reviewState;` is incorrect now, because the server would reject such patch.

You're right. I just thought it's limitation we plan to remove in long term. Also, such change could be done by palindrom, as well.

Anyway, I think what answers all my concerns is:
> if it's just a proxy, or just a plain JS object, it will not be deep cloned.

I think we have to document very precisely and clearly when we do deep clone given object.
Now, I feel little bit confused when and why we do so, so it's hard for me to judge.

So speaking of 
>  to only worry about this if we actually face an issue because of it

I would like to make sure we also face the issue for which we do deep clone at all.